packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/cap.dart

```diff
@@ -6,27 +6,54 @@ import 'package:flutter/foundation.dart' show immutable;
 
 import 'types.dart';
 
+/// Enumeration of possible types of caps.
+enum CapType {
+  /// Cap that is squared off exactly at the start or end vertex of a [Polyline]
+  /// with solid stroke pattern, equivalent to having no additional cap beyond
+  /// the start or end vertex.
+  butt('buttCap'),
```

Comment:
Since the only use case for the strings is `toJson`, and we want to discourage people relying on strings in the interface of this package, I would much prefer that these be completely internal to the implementation of `toJson` rather than public API, so we aren't effectively expanding our legacy API surface.